File: packages/api-models-ts/src/Weather.ts
/**
 * Request and response models for `/api/weather`.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * Cached OpenWeather data for a telescope site (upstream `Weather`).
 *
 * The handler builds this dict by hand rather than serializing the model:
 * `weather` is the raw OpenWeather `weather_info` JSON blob,
 * `weather_retrieved_at` is the model's `retrieved_at`, and the remaining keys
 * are copied off the associated `Telescope`.
 *
 * @since 1.0.0
 * @category Models
 */
export const Weather = Schemas.model(
    v.strictObject({
        weather: Schemas.nullish(Schemas.JsonObject),
        weather_retrieved_at: Schemas.NullishTimestamp,
        weather_fetch_at: Schemas.NullishTimestamp,
        weather_link: Schemas.NullishString,
        telescope_name: Schemas.NullishString,
        telescope_nickname: Schemas.NullishString,
        telescope_id: Schemas.NullishInteger,
        message: Schemas.NullishString,
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type Weather = v.InferOutput<typeof Weather>;

File: packages/api-models-py/skyportal_py_models/weather.py
"""Request and response models for ``/api/weather``."""

from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class Weather(BaseModel):
    """Cached OpenWeather data for a telescope site (upstream ``Weather``).

    The handler builds this dict by hand rather than serializing the model:
    ``weather`` is the raw OpenWeather ``weather_info`` JSON blob,
    ``weather_retrieved_at`` is the model's ``retrieved_at``, and the
    remaining keys are copied off the associated ``Telescope``.
    """

    model_config = ConfigDict(extra="forbid")

    weather: dict[str, Any] | None = None
    weather_retrieved_at: datetime | None = None
    weather_fetch_at: datetime | None = None
    weather_link: str | None = None
    telescope_name: str | None = None
    telescope_nickname: str | None = None
    telescope_id: int | None = None
    message: str | None = None
